controllers/nonadmin_controller.go

```diff
@@ -86,6 +85,21 @@ func (r *DPAReconciler) ReconcileNonAdminController(log logr.Logger) (bool, erro
 		return true, nil
 	}
 
+	selector, err := fields.ParseSelector(fmt.Sprintf("metadata.namespace!=%s", r.NamespacedName.Namespace))
```

Comment:
got `non-exact field matches are not supported by the cache` error with this approach

will change